lib/apiaria.rb

```diff
@@ -28,4 +28,27 @@ def get_mentor_availability(mentor_id)
     JSON.parse(response.body)
   end
   
+  def get_messages(pageNum = nil)
+    unless pageNum
+      response = self.class.get("/message_threads",  headers: { "authorization" => @auth_token })
+      JSON.parse(response.body)
+    else
+      url = "/message_threads?page=" + pageNum.to_s
+      response = self.class.get(url,  headers: { "authorization" => @auth_token })
+      JSON.parse(response.body)
+    end
+  end
+
+  def create_message(sender, recipientId, body, options = {})
```

Comment:
@wonderwoman13 I didnt quite get what you mean by "**setting method arguments to the argument name is not needed here**".

Since _sender_,_recipientId_ and _body_ are mandatory,I used separate arguments for them.And for the optional arguments(_token,subject_),I used an options hash.Is there something wrong with this approach?

Here is how I did i tested it -
2.3.0 :001 > **require './lib/apiaria.rb'** => true 

2.3.0 :002 > **p  = Apiaria.new("praveenyr@gmail.com", "xxxxxx")** 
=> #<Apiaria:0x00000001c83308 @email="praveenyr@gmail.com", @password="xxxxxx", @auth_token="eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhcGlfa2V5IjoiYTBhZTFiZmI0MWI2NGNjNjllMDYzYmI1M2JkM2JjYWYiLCJ1c2VyX2lkIjoyMzQ2ODQxLCJuYW1lIjoiUHJhdmVlbiBZYWxsYWthcmEifQ.0IwoTCsj9_EByW35pU10rlkQWyWJl3w8j5e7MyMcVPk"> 

2.3.0 :003 > **p.create_message("praveenyr@gmail.com", 623967, "I need some Hawaii!",  { token: "c0661d18-7d91-42ac-9f21-99e4ff255994", subject: "Kele API Client"})**  
=>{"message"=>"There was an error processing the message."}